File: java/Arrays and Strings/UniqueCharacters.java
import java.util.*;
import java.lang.*;
import java.io.*;

class UniqueCharacters {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String str = in.nextLine();
        System.out.println(isUniqueChars(str));
        System.out.println(isUniqueChars2(str));
    }

    public static boolean isUniqueChars(String str) {
        boolean char_set[] = new boolean[256];
        for (int i = 0; i < str.length(); i++) {
            int val = str.charAt(i);
            if (char_set[val]) return false;
            char_set[val] = true;
        }
        return true;
    }

    public static boolean isUniqueChars2(String str) {
        int checker = 0;
        for (int i = 0; i < str.length(); i++) {
            int val = str.charAt(i) - 'a';
            if ((checker & (1 << val)) > 0) return false;
            checker |= (1 << val);
        }
        return true;
    }
}

File: js/Arrays and Strings/uniqueCharacters.js
'use strict';

function isUniqueChars(str) {
    var char_set = new Array(256);
    for (var j = 0; i < 256; i++)
        char_set.push(false);

    for (var i = 0; i < str.length; i++) {
        var val = str.charCodeAt(i);
        if (char_set[val]) return false;
        char_set[val] = true;
    }

    return true;
}

function isUniqueChars2(str) {
    var checker = 0;
    for (var i = 0; i < str.length; i++) {
        var val = str.charCodeAt(i) - 97;
        if ((checker & (1 << val)) > 0) return false;
        checker |= (1 << val);
    }
    return true;
}
